lib/vrf.c

```diff
@@ -279,6 +401,50 @@ struct vrf *vrf_lookup_by_id(vrf_id_t vrf_id)
 	return (RB_FIND(vrf_id_head, &vrfs_by_id, &vrf));
 }
 
+/* Finds vrf_id of vrf with table_id or ns_id.
+ *
+ * table_id
+ *    The table_id associated with vrf of interest
+ *
+ * ns_id
+ *    The ns_id associated with vrf of interest
+ *
+ * Description:
+ *    This function checks if vrf is backened by netns
+ *    or vrf lite, and looks them up appropriately.
+ *    Hashmap look up implemented for vrf lite, but not netns.
+ *
+ * Returns:
+ *    vrf_id if found table by table_id, or VRF_DEFAULT if not found
+ */
+vrf_id_t vrf_lookup_by_table(uint32_t table_id, ns_id_t ns_id)
+{
+	struct vrf *vrf;
+	struct zebra_vrf *zvrf;
+
+	/* case vrf with netns : match the netnsid */
+	if (vrf_is_backend_netns()) {
+		RB_FOREACH (vrf, vrf_id_head, &vrfs_by_id) {
+			zvrf = vrf->info;
```

Comment:
This code is only correct for zebra.
`vrf->info` can not be casted to `struct zebra_vrf` in other daemons.

Reply:
I've refactored this in the most recent commit. The cast to `zebra_vrf` stuct is not needed here and is not indicative of daemon-specific functionality.